test/integration/test_http.py

```diff
@@ -73,5 +73,196 @@ def test_basicTxn(self):
+        elif api == "/api/Read":
+            record = {partitionKeyName: partition, rangeKeyName: range_key}
+            request = {"collectionName": coll, "schemaName": schema, "txnID": txn, "schemaVersion": ver, "record": record}
+        elif api == "/api/EndTxn":
+            # Commit
+            request = {"txnID": txn, "commit": True}
+            print(request)
+        else:
+            raise ValueError('Invalid api' + api)
+
+        url = args.http + api
+        r = requests.post(url, data=json.dumps(request))
+        result = r.json()
+        print(result)
+        return result
+
+    def test_schema_validation(self):
+        # Begin Txn
+        data = {}
+        url = args.http + "/api/BeginTxn"
+        r = requests.post(url, data=json.dumps(data))
+        result = r.json()
+        result = result[0][0]
+        print(result)
+        self.assertEqual(result["status"]["code"], 201);
+        ID = result["txnID"]
+
+        # Write with bad Txn ID, should fail
+        result = self.sendReq("/api/Write", ID+1)
+        result = result[0]
+        self.assertEqual(result["status"]["code"], 400)
+
+        # Write with bad collectionName, should fail
+        result = self.sendReq("/api/Write", ID, coll="HTTPClient1")
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Write with bad schemaName, should fail
+        result = self.sendReq("/api/Write", ID, schema="test_schema1")
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Write with bad schemaVersion, should fail
+        result = self.sendReq("/api/Write", ID, ver=2)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Write with bad partition value data type, should fail
+        result = self.sendReq("/api/Write", ID, partition=1)
+        self.assertEqual(result["code"], 500)
+
+        # Write with bad range data type, should fail
+        result = self.sendReq("/api/Write", ID, range_key=1)
+        self.assertEqual(result["code"], 500)
+
+        # Write with bad data field data type, should fail
+        result = self.sendReq("/api/Write", ID, data=1)
+        self.assertEqual(result["code"], 500)
+
+        # Try read data written by failed requests, should fail
+        result = self.sendReq("/api/Read", ID)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 404)
+
+        # Do a valid write, should succeed
+        result = self.sendReq("/api/Write", ID)
+        result = result[0][0]
+        self.assertEqual(result["status"]["code"], 201)
+
+        # Read with bad Txn ID, should fail
+        result = self.sendReq("/api/Read", ID+1)
+        result = result[0]
+        self.assertEqual(result["status"]["code"], 400)
+
+        # Read with bad collectionName, should fail
```

Comment:
Why are some of the tests repeated here?

Reply:
The previous one is for write, this one for read